pkg/aws/ec2/pricing_map.go

```diff
@@ -326,3 +390,46 @@ func ListSpotPrices(ctx context.Context, client ec2client.EC2) ([]ec2Types.SpotP
 	}
 	return spotPrices, nil
 }
+
+func ListStoragePrices(ctx context.Context, region string, client pricingClient.Pricing) ([]string, error) {
+	input := &pricing.GetProductsInput{
+		ServiceCode: aws.String("AmazonEC2"),
+		Filters: []types.Filter{
+			{
+				Field: aws.String("regionCode"),
+				Type:  "TERM_MATCH",
+				Value: aws.String(region),
+			},
+			// Get prices for EBS Volumes
+			{
+				Field: aws.String("productFamily"),
+				Type:  "TERM_MATCH",
+				Value: aws.String("Storage"),
+			},
+		},
+	}
+
+	return getPricesFromProductList(ctx, input, client)
+}
+
+func getPricesFromProductList(ctx context.Context, input *pricing.GetProductsInput, client pricingClient.Pricing) ([]string, error) {
+	var productOutputs []string
+
+	for {
+		products, err := client.GetProducts(ctx, input)
+		if err != nil {
+			return productOutputs, err
+		}
+
+		if products == nil {
+			break
+		}
+
+		productOutputs = append(productOutputs, products.PriceList...)
+		if products.NextToken == nil {
```

Comment:
Should we also check to see if the value is an empty string? This bits confused me quite a few times with aws responses, but I've had to also defensively check `if products.NextToken == nil || *products.NextToken == "" {...}`

Reply:
didn't know why you were doing that. I can add it, doesn't hurt to be extra careful.

Reply:
I'm trying to remember if I actually experienced the error, or if it was when doing tests. I agree it doens't hurt to be extra careful, especially for a loop that makes API requests that can get stuck in an infinite loop 😬

Reply:
Looking at `NextToken`'s type it's a `*string`. Since it's a pointer and not an actual string, it could never be an empty string, no? 🤔

Reply:
done